envisage/plugin.py

```diff
@@ -427,3 +433,11 @@ def factory(**properties):
             return getattr(self, trait_name)
 
         return self.application.register_service(protocol, factory)
+
+    ###########################################################################
+    # 'object' interface.
+    ###########################################################################
+
+    def __repr__(self):
+        """ String representation of a Plugin object """
+        return "Plugin(id={}, name={})".format(self.id, self.name)
```

Comment:
I suggest using `{!r}` in place of `{}` here: as a general rule, `repr`s of objects should be composed of `repr`s of the pieces, and this also ensures that we always get something `eval`-able.

(Same with the `Service` below.)

```suggestion
        return "Plugin(id={!r}, name={!r})".format(self.id, self.name)
```